orangecontrib/text/widgets/owcorpusviewer.py

```diff
@@ -115,7 +115,8 @@ def load_features(self):
         self.selected_features = []
         if self.corpus is not None:
             self.features = [(meta.name, vartype(meta)) for meta in self.corpus.domain.metas] + \
-                            [(attr.name, vartype(attr)) for attr in self.corpus.domain.attributes]
+                            [(attr.name, vartype(attr)) for attr in self.corpus.domain.attributes] + \
+                            [(class_var.name, vartype(class_var)) for class_var in self.corpus.domain.class_vars]
```

Comment:
The behavior we have invalidates the established Python's user contract, namely that the container iterates through its contents:
```py
assert attr in domain == attr in [i for i in domain]
```
The arising confusion should be enough reason alone.

Indeed, the current implementation adapts to classification/regression algorithms, yet the more we become _data mining_ suite (as opposed to simply machine learning suite), the more meta features we will be seeing and manipulating.

Not that they are seldomly used anyway:
```
~/Orange3 $ grep --include='*.py' -nR '\.metas\b' Orange networks datafusion textmining bioinformatics | grep -v '/tests/'
Orange/projection/pca.py:94:             domain.class_vars, domain.metas)
Orange/projection/cur.py:139:            metas = data.metas[self.proj.samples_]
Orange/preprocess/continuize.py:93:        return Domain(new_attrs, new_classes, domain.metas)
Orange/preprocess/remove.py:83:                       for var in domain.metas]
Orange/preprocess/preprocess.py:113:            attributes, data.domain.class_vars, data.domain.metas)
Orange/preprocess/preprocess.py:144:            newattrs, data.domain.class_vars, data.domain.metas)
Orange/preprocess/preprocess.py:170:                                    data.domain.metas)
Orange/preprocess/preprocess.py:171:        return Orange.data.Table(domain, X, data.Y, data.metas)
Orange/preprocess/preprocess.py:194:                                    data.domain.metas)
Orange/preprocess/preprocess.py:349:            self.randomize(new_data.metas)
Orange/preprocess/normalize.py:28:        domain = Domain(new_attrs, new_class_vars, data.domain.metas)
Orange/preprocess/fss.py:88:                                    data.domain.class_vars, data.domain.metas)
Orange/preprocess/fss.py:125:            data.domain.class_vars, data.domain.metas)
Orange/preprocess/fss.py:141:                                    data.domain.metas)
Orange/canvas/report/report.py:535:         ("Meta attributes", bool(domain.metas) and
Orange/canvas/report/report.py:536:          clip_attrs(domain.metas, "meta attributes")),
Orange/canvas/report/report.py:581:    items["Meta attributes"] = len(domain.metas) or "None"
Orange/widgets/settings.py:627:            metas = encode(domain.metas, match == self.MATCH_VALUES_ALL)
Orange/widgets/settings.py:637:        context.metas = metas
Orange/widgets/settings.py:644:                                       for attr in domain.metas]
Orange/widgets/settings.py:705:            get_meta = lambda name: context.metas.get(name, None)
Orange/widgets/settings.py:733:            if not setting.exclude_metas and value in context.metas:
Orange/widgets/settings.py:734:                return value, context.metas[value]
Orange/widgets/settings.py:757:        if (attrs, metas) == (context.attributes, context.metas):
Orange/widgets/settings.py:818:                                                            (c.metas and len(c.metas) or 0)) <= mp)
Orange/widgets/settings.py:826:                           (c.metas and len(c.metas) or 0))
Orange/widgets/settings.py:892:                encode(domain.metas))
Orange/widgets/settings.py:897:            context.attributes, context.class_vars, context.metas) and 2
Orange/widgets/utils/datacaching.py:41:        attrs = data.domain.variables + data.domain.metas
Orange/widgets/utils/datacaching.py:46:        attrs = data.domain.variables + data.domain.metas
Orange/widgets/utils/itemmodels.py:601:            coldesc = make_basket(domain.metas, self.M_density,
Orange/widgets/utils/itemmodels.py:606:                        for var in domain.metas]
Orange/widgets/utils/itemmodels.py:609:        self.vars = domain.attributes + domain.class_vars + domain.metas
Orange/widgets/evaluate/owconfusionmatrix.py:260:            metas = self.data.domain.metas
Orange/widgets/evaluate/owconfusionmatrix.py:281:            M = self.data.metas[row_indices]
Orange/widgets/evaluate/owpredictions.py:199:        metas = list(self.data.domain.metas) + newmetas
Orange/widgets/evaluate/owpredictions.py:209:            predictions.metas[:, -newcolumns.shape[1]:] = newcolumns
Orange/widgets/data/owmergedata.py:171:    return data.domain.attributes + data.domain.class_vars + data.domain.metas
Orange/widgets/data/owmergedata.py:183:    all_vars_A = set(A.domain.variables + A.domain.metas)
Orange/widgets/data/owmergedata.py:186:        ((-i, m) for i, m in enumerate(B.domain.metas, start=1))
Orange/widgets/data/owmergedata.py:265:        left.domain.metas + right.domain.metas
Orange/widgets/data/owmergedata.py:269:    metas = join_array_by_indices(left.metas, right.metas, indices)
Orange/widgets/data/owselectrows.py:120:        for var in chain(self.data.domain.variables, self.data.domain.metas):
Orange/widgets/data/owselectrows.py:143:        for i in range(len(domain.variables) + len(domain.metas)):
Orange/widgets/data/owselectrows.py:284:            len(data.domain.variables) + len(data.domain.metas) > 100)
Orange/widgets/data/owselectrows.py:301:                     data.domain.variables + data.domain.metas]
Orange/widgets/data/owselectrows.py:399:            sp(len(data.domain.variables) + len(data.domain.metas))))
Orange/widgets/data/owselectcolumns.py:444:            all_vars = data.domain.variables + data.domain.metas
Orange/widgets/data/owselectcolumns.py:452:                                for i, attr in enumerate(data.domain.metas)})
Orange/widgets/data/owselectcolumns.py:652:            self.meta_attrs[:] = self.data.domain.metas
Orange/widgets/data/owselectcolumns.py:660:        if (in_domain.attributes, in_domain.class_vars, in_domain.metas) == (
Orange/widgets/data/owselectcolumns.py:661:                out_domain.attributes, out_domain.class_vars, out_domain.metas):
Orange/widgets/data/owselectcolumns.py:665:            diff = list(set(in_domain.variables + in_domain.metas) -
Orange/widgets/data/owselectcolumns.py:666:                        set(out_domain.variables + out_domain.metas))
Orange/widgets/data/owcontinuize.py:251:    return Orange.data.Domain(newattrs, newclass, domain.metas)
Orange/widgets/data/owdiscretize.py:433:            metas=self.data.domain.metas
Orange/widgets/data/owcolor.py:285:                                        create_part(domain.metas))
Orange/widgets/data/owtable.py:816:            if len(colsel) < len(domain) + len(domain.metas):
Orange/widgets/data/owtable.py:818:                allvars = domain.variables + domain.metas
Orange/widgets/data/owtable.py:932:        M_part = parts(table.metas, table.metas_density(), M_dist)
Orange/widgets/data/owtable.py:994:    text += [("%s meta attribute%s" % sp(len(summary.domain.metas)))
Orange/widgets/data/owrank.py:476:                                        metas=self.data.domain.metas)
Orange/widgets/data/owfeatureconstructor.py:189:                                         domain.metas)
Orange/widgets/data/owfeatureconstructor.py:368:                            self.data.domain.variables + self.data.domain.metas]
Orange/widgets/data/owfeatureconstructor.py:549:        source_vars = tuple(self.data.domain) + self.data.domain.metas
Orange/widgets/data/owfeatureconstructor.py:557:            metas=self.data.domain.metas + tuple(metas)
Orange/widgets/data/owgooglesheets.py:121:            len(data), len(data.domain.attributes), len(data.domain.metas))
Orange/widgets/data/owoutliers.py:180:            new_metas = list(self.data.domain.metas) + \
Orange/widgets/data/owoutliers.py:184:            self.new_data.metas = np.hstack((self.data.metas, mahal))
Orange/widgets/data/owconcatenate.py:201:    metas = [table.metas for table in tables]
Orange/widgets/data/owconcatenate.py:223:        tuple(unique(A.metas + B.metas))
Orange/widgets/data/owconcatenate.py:236:        tuple_intersection(A.metas, B.metas),
Orange/widgets/data/owconcatenate.py:252:        domain.metas + new_metas
Orange/widgets/data/owconcatenate.py:267:    metas = numpy.hstack((data.metas,) + tuple(metas))
Orange/widgets/data/owimpute.py:377:                attrs, class_vars, self.data.domain.metas)
Orange/widgets/data/oweditdomain.py:451:        self.input_vars = tuple(domain) + domain.metas
Orange/widgets/data/owdatainfo.py:108:        if not domain.metas:
Orange/widgets/data/owdatainfo.py:111:            disc_metas = count(domain.metas, DiscreteVariable)
Orange/widgets/data/owdatainfo.py:112:            cont_metas = count(domain.metas, ContinuousVariable)
Orange/widgets/data/owdatainfo.py:113:            str_metas = count(domain.metas, StringVariable)
Orange/widgets/data/owdatainfo.py:172:        dd["Meta attributes"] = len(domain.metas) > 0 and join_if((
Orange/widgets/data/owfile.py:16:    vars = examples.domain.variables + examples.domain.metas
Orange/widgets/data/owfile.py:428:                len(data), len(domain.attributes), len(domain.metas))
Orange/widgets/data/owpreprocess.py:603:                                    data.domain.metas)
Orange/widgets/data/owimageviewer.py:416:            self.allAttrs = domain.variables + domain.metas
Orange/widgets/visualize/owheatmap.py:44:        group_domain = Domain(attrs, domain.class_vars, domain.metas)
Orange/widgets/visualize/owheatmap.py:680:                       data.domain.metas),
Orange/widgets/visualize/owheatmap.py:693:            variables = self.data.domain.class_vars + self.data.domain.metas
Orange/widgets/visualize/owheatmap.py:842:                        self.input_data.domain.metas),
Orange/widgets/visualize/owscatterplot.py:277:            new_attrs = [a for a in data.domain.attributes + data.domain.metas
Orange/widgets/visualize/owscatterplot.py:279:            new_metas = [m for m in data.domain.metas if not m.is_primitive()]
Orange/widgets/visualize/owscatterplot.py:332:        for var in self.data.domain.metas:
Orange/widgets/visualize/owscatterplot.py:344:        for var in self.data.domain.metas:
Orange/widgets/visualize/owvenndiagram.py:643:        metas.extend(v for v in table.domain.metas
Orange/widgets/visualize/owvenndiagram.py:646:        variables_seen.update(table.domain.metas)
Orange/widgets/visualize/owvenndiagram.py:754:    for meta in table.domain.metas:
Orange/widgets/visualize/owvenndiagram.py:771:        for var in domain.variables + domain.metas:
Orange/widgets/visualize/owvenndiagram.py:822:    all_possible = [var for var in table.domain.variables + table.domain.metas
Orange/widgets/visualize/owvenndiagram.py:874:    return [attr for attr in domain.variables + domain.metas
Orange/widgets/visualize/owvenndiagram.py:882:    return [attr for attr in domain.variables + domain.metas
Orange/widgets/visualize/owvenndiagram.py:1550:    X, Y, M, W = data.X, data.Y, data.metas, data.W
Orange/widgets/visualize/owvenndiagram.py:1554:    metas = domain.metas
Orange/widgets/visualize/owvenndiagram.py:1582:        filter_vars(data.domain.metas)
Orange/widgets/unsupervised/owmds.py:422:            all_vars = list(domain.variables + domain.metas)
Orange/widgets/unsupervised/owmds.py:938:            metas = domain.metas
Orange/widgets/unsupervised/owmds.py:955:                output.metas[:, -2:] = embedding.X
Orange/widgets/unsupervised/owmds.py:1044:    if variable in data.domain.metas:
Orange/widgets/unsupervised/owmds.py:1045:        x = numpy.array(x.metas, dtype='float').ravel()
Orange/widgets/unsupervised/owhierarchicalclustering.py:977:                items.domain.metas,
Orange/widgets/unsupervised/owhierarchicalclustering.py:978:                [model.Separator] if items.domain.class_vars or items.domain.metas else [],
Orange/widgets/unsupervised/owhierarchicalclustering.py:1135:                metas = domain.metas
Orange/widgets/unsupervised/owhierarchicalclustering.py:1159:                items.domain.class_vars, items.domain.metas)
Orange/widgets/unsupervised/owhierarchicalclustering.py:1163:                items.domain.class_vars, items.domain.metas)
Orange/widgets/unsupervised/owdistancemap.py:442:                        list(items.domain) + list(items.domain.metas))
Orange/widgets/unsupervised/owdistancemap.py:624:                    self.items.domain.metas)
Orange/widgets/unsupervised/owkmeans.py:327:        meta_attrs = domain.metas
Orange/widgets/unsupervised/owpca.py:334:                self.data.domain.metas
Orange/evaluation/testing.py:260:        new_meta_attr = list(data.domain.metas) + new_meta_attr
Orange/evaluation/testing.py:261:        new_meta_vals = np.hstack((data.metas, new_meta_vals))
Orange/evaluation/testing.py:357:        metas = data.metas.copy()
Orange/distance/__init__.py:16:                             table.domain.metas)
Orange/data/instance.py:28:            self._metas = np.array([var.Unknown for var in domain.metas],
Orange/data/instance.py:80:        + len(self.domain.class_vars) + len(self.domain.metas)`.
Orange/data/instance.py:146:        if self._domain.metas:
Orange/data/instance.py:148:                 self.str_values(self._metas, self._domain.metas, limit) + \
Orange/data/sql/table.py:128:                elif var in type_hints.metas:
Orange/data/sql/table.py:227:        attributes = self.domain.variables + self.domain.metas
Orange/data/sql/table.py:239:        attributes = self.domain.variables + self.domain.metas
Orange/data/sql/table.py:336:        metas = [np.empty((0, len(self.domain.metas)))]
Orange/data/sql/table.py:405:                columns += list(self.domain.metas)
Orange/data/table.py:59:        self._metas = table.metas[row_index]
Orange/data/table.py:105:                self.table.metas[self.row_index, -1 - key] = value
Orange/data/table.py:129:        if self._domain.metas:
Orange/data/table.py:130:            s += " {" + sp_values(table.metas, domain.metas) + "}"
Orange/data/table.py:142:        for v in chain(domain, domain.metas):
Orange/data/table.py:231:        self.metas = np.empty((n_rows, len(self.domain.metas)), object)
Orange/data/table.py:266:                arr = _subarray(source.metas, row_indices,
Orange/data/table.py:286:                    a[:, i] = source.metas[row_indices, -1 - col]
Orange/data/table.py:323:            if any(isinstance(var, StringVariable) for var in domain.metas):
Orange/data/table.py:325:            self.metas = get_columns(row_indices, conversion.metas,
Orange/data/table.py:327:            if self.metas.ndim == 1:
Orange/data/table.py:328:                self.metas = self.metas.reshape(-1, len(self.domain.metas))
Orange/data/table.py:362:        self.metas = source.metas[row_indices]
Orange/data/table.py:363:        if self.metas.ndim == 1:
Orange/data/table.py:364:            self.metas = self.metas.reshape(-1, len(self.domain.metas))
Orange/data/table.py:421:        if metas.shape[1] != len(domain.metas):
Orange/data/table.py:424:                    metas.shape[1], len(domain.metas))
Orange/data/table.py:434:        self.metas = metas
Orange/data/table.py:446:        metas = domain.metas
Orange/data/table.py:456:                self.metas[i, j] = var.to_val(val)
Orange/data/table.py:592:                self.metas[row] = example._metas
Orange/data/table.py:596:            self.X[row], self._Y[row], self.metas[row] = \
Orange/data/table.py:612:            self.metas[row] = np.array([var.Unknown for var in domain.metas],
Orange/data/table.py:631:            self.metas.resize(new_length, self.metas.shape[1])
Orange/data/table.py:642:            if self.metas.shape[0] == new_length:
Orange/data/table.py:643:                self.metas.resize(old_length, self.metas.shape[1])
Orange/data/table.py:675:                    return Value(var, self.metas[row_idx, -1 - col_idx])
Orange/data/table.py:741:                    self.metas[row_idx, -1 - col_idx] = value
Orange/data/table.py:757:                    self.metas[row_idx, -1 - col] = var.to_val(value)
Orange/data/table.py:783:                self.metas[row_idx, meta_cols] = value
Orange/data/table.py:792:        self.metas = np.delete(self.metas, key, axis=0)
Orange/data/table.py:840:            self.metas[row + 1:] = self.metas[row:-1]
Orange/data/table.py:850:            self.metas[row:-1] = self.metas[row + 1:]
Orange/data/table.py:874:                self.metas[old_length:] = instances.metas
Orange/data/table.py:910:                               chain(t.domain.variables, t.domain.metas)))
Orange/data/table.py:916:                            flatten(t.domain.metas for t in tables))
Orange/data/table.py:924:                                     np.hstack(tuple(ndmin(t.metas) for t in tables)),
Orange/data/table.py:935:                (not self.metas.shape[-1] or self.metas.base is not None) and
Orange/data/table.py:944:                (self.metas.base is None) and
Orange/data/table.py:955:        if self.metas.base is not None:
Orange/data/table.py:956:            self.metas = self.metas.copy()
Orange/data/table.py:994:            self._metas_density = Table.__determine_density(self.metas)
Orange/data/table.py:1034:            cs = zlib.adler32(np.ascontiguousarray(self.metas), cs)
Orange/data/table.py:1046:        self.metas = self.metas[ind]
Orange/data/table.py:1074:            return rx(self.metas[:, -1 - index])
Orange/data/table.py:1242:            if include_metas and self.domain.metas:
Orange/data/table.py:1243:                rr.append(bn.stats(self.metas, W))
Orange/data/table.py:1251:            ms = any(c < 0 for c in columns) and bn.stats(self.metas, W)
Orange/data/table.py:1337:            row_data = self.metas[:, -1 - row_indi]
Orange/data/table.py:1367:                (self.metas, lambda i: i < 0, lambda i: -1 - i)):
Orange/data/domain.py:57:        self.metas = [
Orange/data/domain.py:59:            else var.compute_value for var in destination.metas]
Orange/data/domain.py:116:        self.metas = tuple(metas + append_metas)
Orange/data/domain.py:126:            for idx, var in enumerate(self.metas)))
Orange/data/domain.py:210:        return len(self.attributes) + len(self.class_vars) + len(self.metas)
Orange/data/domain.py:230:            return self.metas[-1-idx]
Orange/data/domain.py:354:                     for i in c.metas]
Orange/data/domain.py:423:                self.metas == other.metas)
Orange/data/domain.py:426:        return hash(self.attributes) ^ hash(self.class_vars) ^ hash(self.metas)
Orange/data/io.py:477:                                      data.domain.metas)]
Orange/data/io.py:490:                                           data.domain.metas)]
Orange/data/io.py:499:                                                  zip(repeat('meta'), data.domain.metas)))))
Orange/data/io.py:514:                          data.domain.metas))
Orange/data/io.py:518:                       data.metas):
Orange/datasets/list_update.py:25:            'meta': len(domain.metas),
networks/orangecontrib/network/network.py:169:                          self._items.domain.metas))
networks/orangecontrib/network/network.py:175:                                 self._items.domain.metas)
networks/orangecontrib/network/widgets/OWNxExplorer.py:552:                                                    items.domain.metas))
networks/orangecontrib/network/widgets/OWNxExplorer.py:554:                                                      domain.metas)))
networks/orangecontrib/network/widgets/OWNxExplorer.py:632:        elif attribute in table.domain.metas:
networks/orangecontrib/network/widgets/OWNxExplorer.py:633:            values = table[:, attribute].metas[:, 0]
networks/orangecontrib/network/widgets/OWNxExplorer.py:657:        elif attribute in table.domain.metas:
networks/orangecontrib/network/widgets/OWNxExplorer.py:658:            values = table[:, attribute].metas[:, 0]
networks/orangecontrib/network/widgets/OWNxFile.py:190:                                       items.domain.metas)
networks/orangecontrib/network/community.py:30:                        items.domain.metas)
datafusion/orangecontrib/datafusion/widgets/owtabletorelation.py:94:            self.init_attr_values(data.domain.metas)
datafusion/orangecontrib/datafusion/widgets/owtabletorelation.py:125:            self.row_names = list(self.data[:, -self.selected_meta].metas.flatten())
datafusion/orangecontrib/datafusion/widgets/owtabletorelation.py:159:            metadata_cols = list(domain.class_vars) + list(domain.metas)
datafusion/orangecontrib/datafusion/models.py:29:        meta_vars, self.metas = self._create_metas(relation)
textmining/orangecontrib/text/lda.py:64:                        metas=corp_in.domain.metas)
textmining/orangecontrib/text/lda.py:69:                                metas=corp_in.metas)
textmining/orangecontrib/text/widgets/owloadcorpus.py:136:            for i in self.corpus.domain.metas:
textmining/orangecontrib/text/widgets/owbagofwords.py:113:            domain = Domain(attr, self.corpus.domain.class_vars, metas=self.corpus.domain.metas)
textmining/orangecontrib/text/widgets/owbagofwords.py:116:            new_table = Table.from_numpy(domain, freqs, Y=self.corpus._Y, metas=self.corpus.metas)
textmining/orangecontrib/text/widgets/owwordcloud.py:197:        metas = data.domain.metas if data else []
textmining/orangecontrib/text/widgets/owwordcloud.py:203:            words = data.metas[:, col]
textmining/orangecontrib/text/widgets/owwordcloud.py:204:            self.n_topic_words = data.metas.shape[0]
textmining/orangecontrib/text/widgets/owwordcloud.py:220:            col = next(i for i,var in enumerate(self.corpus.domain.metas) if var.is_string)
textmining/orangecontrib/text/widgets/owwordcloud.py:222:            col = next(i for i,var in enumerate(self.corpus.domain.metas)
textmining/orangecontrib/text/widgets/owwordcloud.py:231:        texts = self.corpus.metas[:, col]
textmining/orangecontrib/text/widgets/owcorpusviewer.py:122:                             for var in chain(domain.variables, domain.metas)]
textmining/orangecontrib/text/widgets/owgeomap.py:56:        attr = self.metas[self.selected_attr]
textmining/orangecontrib/text/widgets/owgeomap.py:108:        self.metas = [a for a in chain(data.domain.metas,
textmining/orangecontrib/text/widgets/owgeomap.py:116:        for i, var in enumerate(self.metas):
textmining/orangecontrib/text/widgets/owgeomap.py:121:        if self.metas:
textmining/orangecontrib/text/widgets/owgeomap.py:122:            self.attr_combo.setCurrentIndex(self.attr_combo.findText(self.metas[self.selected_attr].name))
textmining/orangecontrib/text/widgets/owgeomap.py:157:        attr = self.metas[self.selected_attr]
textmining/orangecontrib/text/corpus.py:31:        self.metas = metas
textmining/orangecontrib/text/corpus.py:57:        for i, attr in enumerate(table.domain.metas):
textmining/orangecontrib/text/corpus.py:67:        for line in range(table.metas.shape[0]):
textmining/orangecontrib/text/corpus.py:68:            documents.append(' '.join(table.metas[line, include_ids]))
textmining/orangecontrib/text/corpus.py:70:        corp = cls(documents, table.X, table.Y, table.metas, table.domain)
textmining/orangecontrib/text/corpus.py:71:        corp.used_features = [f for i, f in enumerate(table.domain.metas) if i in include_ids]
textmining/orangecontrib/text/corpus.py:76:        indices = [self.domain.metas.index(f) for f in selected_features]
textmining/orangecontrib/text/corpus.py:77:        for line in range(self.metas.shape[0]):
textmining/orangecontrib/text/corpus.py:78:            documents.append(' '.join(self.metas[line, indices]))
textmining/orangecontrib/text/corpus.py:83:        self.metas = np.vstack((self.metas, metadata))
textmining/orangecontrib/text/corpus.py:102:                np.array_equal(self.metas, other.metas) and
bioinformatics/orangecontrib/bio/widgets3/OWGeneNetwork.py:278:    variables = domain.variables + domain.metas
bioinformatics/orangecontrib/bio/widgets3/OWGeneNetwork.py:489:        meta = numpy.empty((0, len(nodedomain.metas)),
bioinformatics/orangecontrib/bio/widgets3/OWFeatureSelection.py:1236:                                        domain.metas + (score_var,))
bioinformatics/orangecontrib/bio/widgets3/OWFeatureSelection.py:1251:                selected_attrs, domain.class_vars, domain.metas)
bioinformatics/orangecontrib/bio/widgets3/OWFeatureSelection.py:1255:                remaining_attrs, domain.class_vars, domain.metas)
bioinformatics/orangecontrib/bio/widgets3/OWGEODatasets.py:588:                for meta in data.domain.metas:
bioinformatics/orangecontrib/bio/widgets3/OWGEODatasets.py:607:                data.domain.metas
bioinformatics/orangecontrib/bio/widgets3/OWGenExpress.py:836:                attributes, table.domain.class_vars, table.domain.metas)
bioinformatics/orangecontrib/bio/widgets3/OWMAPlot.py:244:        attrs = [attr for attr in data.domain.variables + data.domain.metas
bioinformatics/orangecontrib/bio/widgets3/OWMAPlot.py:510:                self.data.domain.metas + tuple(newmetas))
bioinformatics/orangecontrib/bio/widgets3/OWMAPlot.py:521:            data.metas[:, -i - 1] = mcol
bioinformatics/orangecontrib/bio/widgets3/OWMAPlot.py:529:                                        data.domain.metas)
bioinformatics/orangecontrib/bio/widgets3/OWGOEnrichmentAnalysis.py:368:            allvars = domain.variables + domain.metas
bioinformatics/orangecontrib/bio/widgets3/OWGOEnrichmentAnalysis.py:584:                              self.referenceDataset.domain.metas):
bioinformatics/orangecontrib/bio/widgets3/OWGOEnrichmentAnalysis.py:863:                self.clusterDataset.domain.metas)
bioinformatics/orangecontrib/bio/widgets3/OWGOEnrichmentAnalysis.py:879:                    self.clusterDataset.domain.metas)
bioinformatics/orangecontrib/bio/widgets3/OWPIPAx.py:914:                attributes, table.domain.class_var, table.domain.metas)
bioinformatics/orangecontrib/bio/widgets3/OWGenotypeDistances.py:309:            domain = Orange.data.Domain(attrs, [], self.data.domain.metas)
bioinformatics/orangecontrib/bio/widgets3/OWGenotypeDistances.py:473:                                        self.data.domain.metas)
bioinformatics/orangecontrib/bio/widgets3/utils/group.py:33:    disc_vars = [var for var in data.domain.class_vars + data.domain.metas
bioinformatics/orangecontrib/bio/widgets3/OWSelectGenes.py:920:    return domain.variables + domain.metas
bioinformatics/orangecontrib/bio/widgets3/OWGeneInfo.py:381:                [attr for attr in data.domain.variables + data.domain.metas
bioinformatics/orangecontrib/bio/widgets3/OWGeneInfo.py:551:                attrs, self.data.domain.class_vars, self.data.domain.metas)
bioinformatics/orangecontrib/bio/widgets3/OWGeneInfo.py:569:                self.data.domain.metas + tuple(metas))
bioinformatics/orangecontrib/bio/widgets3/OWKEGGPathwayBrowser.py:419:            vars = data.domain.variables + data.domain.metas
bioinformatics/orangecontrib/bio/widgets3/OWSetEnrichment.py:396:            varlist = [var for var in data.domain.variables + data.domain.metas
bioinformatics/orangecontrib/bio/widgets3/OWSetEnrichment.py:938:                mapped, self.data.domain.class_vars, self.data.domain.metas)
bioinformatics/orangecontrib/bio/widgets3/OWDisplayProfiles.py:214:            variables = data.domain.variables + data.domain.metas
bioinformatics/orangecontrib/bio/widgets3/OWVulcanoPlot.py:698:                attrs, self.data.domain.class_vars, self.data.domain.metas)
bioinformatics/orangecontrib/bio/dicty/__init__.py:761:        domain = create_domain([et.domain[id] for id in ids], None, et.domain.metas if OR3 else et.domain.getmetas())
bioinformatics/orangecontrib/bio/dicty/__init__.py:1695:    ndom = create_domain(natts, data.domain.class_var, data.domain.metas if OR3 else data.domain.getmetas())
bioinformatics/orangecontrib/bio/utils/compat.py:40:        metaatts = domain.metas
bioinformatics/orangecontrib/bio/utils/compat.py:66:        return domain.metas
```
Zipping/iterating through `domain.vars` is not nearly as "expensive" as chaining or joining or branching for metas is.

Will submit a PR shortly on how I see it better served.